### PR TITLE
Add flow_control_pin

### LINKED ISSUE #125: Support RS485-to-TTL modules with control flow pin
Modules with automatic flow control such as HW-0519 are already supported. MAX485 are currently not supported. Search for `flow_control_pin` in https://github.com/syssi/esphome-solax-x1-mini for what needs to be changed.

### PR DESCRIPTION
**Pull request by tronikos**
Fixes #125
Based on https://github.com/syssi/esphome-solax-x1-mini/pull/19
Will add schematics and instructions from that PR in the wiki after this is merged

### CODE REVIEW

**Comment by barndawgie** [comment]
Do you have the hardware to test this? Or do we need to get someone to test against this branch?

**Comment by tronikos** [comment]
No I don't have the hardware to test this. I don't think it's necessary to have someone to test against this branch. It would be easier for them to test against the main branch and reopen the attached issue if this doesn't work.

File: components/econet/econet.cpp [hunk 1]
```diff
@@ -424,8 +431,16 @@ void Econet::transmit_message_(uint8_t command, const std::vector<uint8_t> &data
   tx_message_.push_back(crc);
   tx_message_.push_back(crc >> 8);
 
+  if (this->flow_control_pin_ != nullptr) {
+    this->flow_control_pin_->digital_write(true);
+  }
+
   this->write_array(&tx_message_[0], tx_message_.size());
-  // this->flush();
```
Comment (barndawgie) [inline components/econet/econet.cpp:428]: Uncommenting this seems to change behavior even when a flow control pin is not added; is that intentional?

**Review by barndawgie — COMMENTED** [review]
Do you plan to provide updated yaml config in a future PR?

File: components/econet/econet.cpp (hunk 1 above)
Comment (tronikos) [reply components/econet/econet.cpp:428]: I could move it in the condition below but it's better to flush for both cases. It's good to immediately flush to ensure all bytes are transmitted together and avoid collisions with other messages, incoming or outgoing.

**Comment by tronikos** [comment]
I plan to mention in the wiki that if you have a module with flow control pin you need to add the following in your yaml:

```
econet:
  flow_control_pin: GPIO0
```

matching the PIN the control flow was connected to.

**Review by barndawgie — APPROVED** [review]
LGTM

**Pull request merged**